addons/binding/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/handler/HyperionHandler.java

```diff
@@ -0,0 +1,315 @@
+        Optional<Transform> defaultTransform = transform.stream() // convert list to stream
+                .findFirst();
+
+        if (defaultTransform.isPresent()) {
+
+            double luminanceGain = defaultTransform.get().getLuminanceGain();
+            if (luminanceGain >= 0.0 && luminanceGain <= 1.0) {
+                PercentType luminanceGainPercentType = new PercentType((int) (luminanceGain * 100));
+                updateState(CHANNEL_BRIGHTNESS, luminanceGainPercentType);
+            }
+        } else {
+            updateState(CHANNEL_BRIGHTNESS, UnDefType.NULL);
+        }
+    }
+
+    private void updateEffect(List<ActiveEffect> activeEffects, List<Effect> effects) {
+        Optional<ActiveEffect> effect = activeEffects.stream() // convert list to stream
+                .findFirst();
+        if (effect.isPresent()) {
+            String path = effect.get().getScript();
+            Optional<Effect> effectDescription = effects.stream() // convert list to stream
+                    .filter(effectCandidate -> effectCandidate.getScript().equals(path)).findFirst();
+            if (effectDescription.isPresent()) {
+                String effectName = effectDescription.get().getName();
+                updateState(CHANNEL_EFFECT, new StringType(effectName));
+            }
+        } else {
+            updateState(CHANNEL_EFFECT, UnDefType.NULL);
+        }
+    }
+
+    private void updateColor(List<ActiveLedColor> activeColors) {
+        Optional<ActiveLedColor> color = activeColors.stream() // convert list to stream
+                .findFirst();
+        if (color.isPresent()) {
+            List<Integer> rgbVals = color.get().getRGBValue();
+            int r = rgbVals.get(0);
+            int g = rgbVals.get(1);
+            int b = rgbVals.get(2);
+            HSBType hsbType = HSBType.fromRGB(r, g, b);
+            updateState(CHANNEL_COLOR, hsbType);
+        } else {
+            updateState(CHANNEL_COLOR, UnDefType.NULL);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Hyperion thing handler.");
+        try {
+            Configuration config = thing.getConfiguration();
+            address = (String) config.get(PROP_HOST);
+            port = ((BigDecimal) config.get(PROP_PORT)).intValue();
+            refreshInterval = ((BigDecimal) config.get(PROP_POLL_FREQUENCY)).intValue();
+            priority = ((BigDecimal) config.get(PROP_PRIORITY)).intValue();
+
+            connection = new JsonTcpConnection(address, port);
+            connectFuture = scheduler.scheduleWithFixedDelay(connectionJob, 0, refreshInterval, TimeUnit.SECONDS);
+            refreshFuture = scheduler.scheduleWithFixedDelay(refreshJob, 0, refreshInterval, TimeUnit.SECONDS);
+        } catch (UnknownHostException e) {
+            logger.warn("Could not resolve host: {}", e.getMessage());
+            updateOnlineStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
maybe add the exception message as a third parameter here

Reply:
done